hs_core/hydroshare/resource.py

```diff
@@ -705,9 +705,10 @@ def delete_resource_file(pk, filename_or_id, user):
     else:
         raise ObjectDoesNotExist(filename_or_id)
 
-    if resource.raccess.public:
+    if resource.raccess.public or resource.raccess.discoverable:
         if not resource.can_be_public:
             resource.raccess.public = False
+            resource.raccess.discoverable = False
```

Comment:
It looks like we have the exact same rule for can_be_public and can_be_discoverable which means we can use the same check criteria for both public and discoverable, right? If this is indeed true, it is better to add a comment to "can_be_public()" method to indicate so or make the method name to be more informative to indicate so.